### PR TITLE
Respect time unit parameter in FunctionTimer.totalTime()

### PR DESCRIPTION
**Pull request by izeye**
This PR changes to respect time unit parameter in `FunctionTimer.totalTime()`.

### CODE REVIEW

File: micrometer-core/src/test/java/io/micrometer/core/instrument/cumulative/CumulativeFunctionTimerTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+class CumulativeFunctionTimerTest {
+
+    @Test
+    void totalTimeWhenStateObjectChangedToNullShouldWorkWithChangedTimeUnit() {
+        CumulativeFunctionTimer functionTimer = new CumulativeFunctionTimer(
+            null, new Object(), (o) -> 1L, (o) -> 1d, TimeUnit.SECONDS, TimeUnit.SECONDS);
+        assertThat(functionTimer.totalTime(TimeUnit.SECONDS)).isEqualTo(1d);
+        assertThat(functionTimer.totalTime(TimeUnit.MILLISECONDS)).isEqualTo(1000d);
+        System.gc();
```
Comment (OleksiiBondar) [inline micrometer-core/src/test/java/io/micrometer/core/instrument/cumulative/CumulativeFunctionTimerTest.java]: Just for learning purposes, why do we need that?

Comment (izeye) [reply micrometer-core/src/test/java/io/micrometer/core/instrument/cumulative/CumulativeFunctionTimerTest.java]: @OleksiiBondar To make the object referenced by the `WeakReference` in the function timer be garbage-collected, hence changing it to `null`.

**Comment by shakuzen** [comment]
Thanks @izeye! How did you manage to find this bug, btw?

**Comment by izeye** [comment]
@shakuzen I noticed when I was looking into #1161.

**Pull request merged**